### PR TITLE
backport `main`  latest changes into `track/2.31`

### PR DESCRIPTION
**Pull request by DnPlas**
The changes this PR is trying to merge should not affect any dex-auth functionality, they are mainly CI changes.

### CODE REVIEW

**Review by misohu — APPROVED** [review]
LGTM :)

File: metadata.yaml [hunk 1]
```diff
@@ -4,6 +4,10 @@ name: dex-auth
 summary: A federated OpenID Connect provider
 description: |
   Dex is an identity service that uses OpenID Connect to drive authentication for other apps.
+website: https://charmhub.io/dex-auth
+source: https://github.com/canonical/dex-auth-operator
+issues: https://github.com/canonical/dex-auth-operator/issues
+docs: https://discourse.charmhub.io/t/charmed-dex/8214
```
Comment (beliaev-maksim) [inline metadata.yaml:10]: do we have a page on https://charmed-kubeflow.io ?

**Review by beliaev-maksim — APPROVED** [review]
all the rest LGTM

File: metadata.yaml (hunk 1 above)
Comment (DnPlas) [reply metadata.yaml:10]: Uh, if you mean something like https://charmed-kubeflow.io//dex-auth, no we do not. We have a list of all the components [here](https://charmed-kubeflow.io/docs/kubeflow-bundle#heading--components), but I don't think that links belongs in here.

**Pull request merged**